workflows/validate.py

```diff
@@ -0,0 +1,78 @@
+"""
+Runs the json validator on a json file and raises an exception if there is an error
+Intended to be used as a hook for repositories which are developing recipes
+
+Example of how this could be used in a .pre-commit-config.yaml file:
+```
+# Validate json recipes
+- repo: https://github.com/DiamondLightSource/python-workflows
+  hooks:
+  - name: Recipe validation
+    id: recipe-validation
+    language: python
+    files: ^recipes/
+    entry: workflows-recipe-validate
+```
+"""
+
+import logging
+import workflows.recipe
+import workflows
+import sys
+import json
+import argparse
+
+
+def validate_recipe(json_filename):
+    """Reads a json file, tries to turn it into a recipe and then validates it.
+    Exits on exception with non-zero error"""
+
+    # Read in the file
+    try:
+        with open(json_filename) as f:
+            recipe_text = f.read()
+    except Exception as e:
+        logging.exception("Could not read recipe from {0}".format(json_filename))
+        raise e
+
+    # Turn it into a recipe and validate
+    try:
+        # Create a recipe object and validate it
+        workflows.recipe.Recipe(recipe_text).validate()
+    except json.decoder.JSONDecodeError as e:
+        logging.error(
+            "JSON error in recipe {0}, please address this".format(json_filename)
+        )
+        logging.error("{0} at line {1} col {2}".format(e.msg, e.lineno, e.colno))
```

Comment:
This information is already in the exception information, so pass that to the logger instead:
`logging.error("blablabla", exc_info=True)`
In general don't generate multiple messages where you can get away with a single one. This makes little difference on the command line, but were you to log this to graylog then you would get 2 separate entries.

"please address this" is superfluous.

Reply:
The reason I did this is that it provides neater output when using as a hook, with just two formatted lines. Otherwise the terminal gets clogged up with traceback.

Reply:
ah yes, ignore what I said about `exc_info` - that is about exception provenance, but you mean the exception contents. Just concatenate the lines using a `\n` and you have a 2-line output that comes as a single log message.

Reply:
Will do this for all of them.